### DISCUSSION TITLE
How to add custom model paramater in tune_model() using kwargs?

### DISCUSSION THREAD

**Discussion by krisvive** [pycaret/pycaret#1217] [Q&A] [answered]
Hi Team,

I've built an XGBoost model using create_model() and next I want to tune it to a specific _scale_pos_weight_ paramater (it's a specific Hyperparameter of XGBoost). I am get different errors while I am using kwargs option of tune_model. 

`tuned_xgb = tune_model(xgb_created, optimize = 'f1',scale_pos_weight=5)`

I get below error...
`TypeError: __init__() got an unexpected keyword argument 'scale_pos_weight'`

I guess I am not using this functionality correct, I would very much appreciate help from anyone who has used it extensively
I've installed Pycaret from 'git+https://github.com/pycaret/pycaret.git' 
xgboost version: 1.4.1

**Comment by Yard1** [comment, marked as answer]
The kwargs in `tune_model` are passed to the searcher, not the model itself. You should rather pass the kwarg to `create_model` like this:
```python
xgb_created = create_model("xgboost", scale_pos_weight=5)
tuned_xgb = tune_model(xgb_created, optimize = 'f1')
```
Let me know if this works for you.

**Reply by krisvive** [reply]
Thanks @Yard1 that worked
